### PR TITLE
CATTY-346 Redesign Upload ViewController

### PR DESCRIPTION
**Pull request by DhrubojyotiBis1**
Redesign Upload ViewController UI.

Screenshot 2020-07-14 at 12 17 54 AM

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by DhrubojyotiBis1** [comment]
Jenkins, retest this please.

File: src/Catty/ViewController/Upload/UploadViewController.swift [hunk 1]
```diff
@@ -81,62 +84,74 @@ class UploadViewController: UIViewController {
...
-        projectNameLabel.textColor = UIColor.globalTint
-        projectNameLabel.text = kLocalizedName
-        projectNameLabel.font = UIFont.boldSystemFont(ofSize: uploadFontSize)
-
-        projectNameTextField.borderStyle = .roundedRect
-        projectNameTextField.layer.borderWidth = 1.0
-        projectNameTextField.layer.borderColor = UIColor.textViewBorderGray.cgColor
-        projectNameTextField.layer.cornerRadius = 3
+        addLineViewElement(withTopConstrain: 3 * verticalConstrainValue, fromElement: self.view)
+        createLabel(text: kLocalizedName, textColor: UIColor.globalTint, font: .boldSystemFont(ofSize: labelFontSize))
```
Comment (m-herold) [inline src/Catty/ViewController/Upload/UploadViewController.swift:108]: Please directly return the created label from this function and replace the "if lets" below with the actual label.

File: src/Catty/ViewController/Upload/UploadViewController.swift [hunk 2]
```diff
@@ -232,6 +257,43 @@ class UploadViewController: UIViewController {
...
     }
 
+    func createLabel(text: String, textColor: UIColor, font: UIFont) {
+        let label = UILabel()
+        label.text = text
+        label.textColor = textColor
+        label.font = font
+        self.view.addSubview(label)
+
+        if textColor == UIColor.globalTint {
```
Comment (m-herold) [inline src/Catty/ViewController/Upload/UploadViewController.swift:267]: Please remove condition based on text color (changes to the color will affect the internal logic which is very error prone). Please make another function `createValue` (or similar).

File: src/Catty/ViewController/Upload/UploadViewController.swift [hunk 3]
```diff
@@ -232,6 +257,43 @@ class UploadViewController: UIViewController {
...
+            if let lastSeperationView = self.seperationView.last {
+             label.topAnchor.constraint(equalTo: lastSeperationView.bottomAnchor, constant: verticalConstrainValue).isActive = true
+            }
+            label.leftAnchor.constraint(equalTo: self.view.leftAnchor, constant: horizontalConstrainValue).isActive = true
+        } else {
+            self.values.append(label)
+        }
+    }
+
+    func addLineViewElement(withTopConstrain topConstrain: CGFloat, fromElement element: UIView) {
```
Comment (m-herold) [inline src/Catty/ViewController/Upload/UploadViewController.swift:281]: Please correct typo (withTopConstraint topConstraint).

**Comment by DhrubojyotiBis1** [comment]
Hi, @m-herold I have made the changes. Please review

File: src/Catty/ViewController/Upload/UploadViewController.swift [hunk 4]
```diff
@@ -81,62 +84,71 @@ class UploadViewController: UIViewController {
...
     func initDescriptionViewElements() {
-        descriptionLabel.textColor = UIColor.globalTint
-        descriptionLabel.text = kLocalizedDescription
-        descriptionLabel.font = UIFont.boldSystemFont(ofSize: uploadFontSize)
+        if let lastLabel = labels.last {
+            addLineViewElement(withTopConstraint: verticalConstrainValue, fromElement: lastLabel)
+        }
+        let descriptionLabel = createLabel(text: kLocalizedDescription, font: .boldSystemFont(ofSize: labelFontSize))
 
+        descriptionTextView = UITextView()
```
Comment (m-herold) [inline src/Catty/ViewController/Upload/UploadViewController.swift:151]: Please change the font size of the `descriptionTextView` to `valueFontSize`.

**Comment by m-herold** [comment]
Thanks!

**Pull request merged**